Branch-SDK/src/main/java/io/branch/referral/network/BranchRemoteInterface.java

```diff
@@ -178,9 +179,14 @@ public static final BranchRemoteInterface getDefaultBranchRemoteInterface(Contex
      * response in Branch SDK terms.
      * see <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html">HTTP/1.1: Status Codes</a>
      */
-    private ServerResponse processEntityForJSON(String responseString, int statusCode, String tag) {
+    private ServerResponse processEntityForJSON(String responseString, int statusCode, String tag,@Nullable String requestId) {
```

Comment:
Can we replace all arguments in `processEntityForJSON` with just `BranchResponse` and `tag`? This way it's clear that `processEntityForJSON` converts from the `BranchResponse` abstraction to json-formatted `ServerResponse`. Also, for brevity, let's get rid of `serverResponse.setRequestId_` and assign that value in the constructor, so it becomes a read-only variable, just like `ServerResponse.statusCode_`. In order to avoid empty strings we can use `Strings.emptyToNull()` when passing the request id to to the `ServerResponse` constructor

Reply:
1. Yeah, let's do Object here, did not wanted change much but will update.
2. Already using `!TextUtils.isEmpty(requestId)` -- this checks for both null and empty. Not sure how using `Strings.emptyToNull()` will help since we have to check for null in any case, the first method does both.

Reply:
> @code-crusher I was waiting on @jdee response regarding the format, but besides that, I think you still haven't addressed the comment about `ServerResponse.requestId`, let me know if doesn't make sense to you, or otherwise, if you disagree with that comment, happy to chat.

Used `Strings.emptyToNull(requestId)` -- just doesn't feel okay to add another step of processing, but it's fine with me.

Reply:
I think we are talking past each other @code-crusher 
> requestId should be a field within ServerResponse the same way it's a field in BranchResponse

I was saying that `ServerResponse` needs to have `requestId` passed in its constructor, as well as `BanchResponse`.. (I don't even know why we have `BanchResponse` in the first place, seems like we could just parse json directly to `ServerResponse`, but we can cross that bridge when we get there.)

Reply:
I don't see why that class needs to have this when BranchResponse is already passing it. What's the benefit of it that I'm missing?

Reply:
> I still think that's the right thing to do because BranchResponse is just an intermediate class, if requestId was ever to be used in some other way besides logging, it would prolly would be referenced via ServerResponse instance which is further downstream.
...
> (I don't even know why we have BanchResponse in the first place, seems like we could just parse json directly to ServerResponse, but we can cross that bridge when we get there.)

Reply:
But that's what I did earlier https://github.com/BranchMetrics/android-branch-deep-linking-attribution/pull/869/commits/f8fa8e3724a88f47b73205a1c1cedf3bdc1fa772 and mentioned the same: https://github.com/BranchMetrics/android-branch-deep-linking-attribution/pull/869#discussion_r513480024

Reply:
Honestly, this is a too small change to think so much on this. I'll add the request id back into ServerResponse and update the PR

Reply:
Sorry for appearing picky, I think we just keep talking past each other.

I know you did that earlier and I agreed that it was a good idea, my [comment](https://github.com/BranchMetrics/android-branch-deep-linking-attribution/pull/869#discussion_r513480024) that you linked to was specifically about removing the setter and instead having the value passed in the constructor of `ServerResponse`.
>get rid of serverResponse.setRequestId_ and assign that value in the constructor, so it becomes a read-only variable, just like ServerResponse.statusCode_